### DISCUSSION TITLE
Use existing resource group

### DISCUSSION THREAD

**Discussion by code-on-git** [actions/virtual-environments#10557] [General questions] [answered]
When I build my images a temporary resource group is created named pkr-Resource-Group-[some random stuff]

Is there any way to create your own resource group to use instaed of the temporary?

**Comment by vidyasagarnimmagaddi** [comment, marked as answer]
Yes @code-on-git  ,you can specify your own resource group when building images with Packer. By default, Packer creates a temporary resource group, but you can set the resource_group_name parameter in your Azure builder configuration to use your own resource group.
Points:

- Ensure that the specified resource group already exists in your Azure account.
- Double-check your Azure permissions, as  service principal needs to have the correct rights to create and manage resources in the specified resource group.
Here is an example:
```
  builder "azure-arm" {
  client_id       = "YOUR_CLIENT_ID"
  client_secret   = "YOUR_CLIENT_SECRET"
  tenant_id       = "YOUR_TENANT_ID"
  subscription_id  = "YOUR_SUBSCRIPTION_ID"
  resource_group_name = "YOUR_CUSTOM_RESOURCE_GROUP"
  image_name      = "YOUR_IMAGE_NAME"
  location        = "YOUR_LOCATION"
  }

**Reply by code-on-git** [reply]
Thank you for your answer. 
But I don't seem to have that option.
Let me describe how I've set this up. Maybe I'm doing it it wrong in some way.

I'm cloning this public runner-images repository from Github to my Azure Devops repository.
Then I'm running a pipeline to execute the GenerateResourcesAndImage command to create the image.

When I search through the entire runner-images repository, I don't find your code anywhere.

Should I run this in some other way instead?
How are you creating your runner-images?

**Comment by Atsushi-6679** [comment]
builder "azure-arm" {
  client_id       = "YOUR_CLIENT_ID"
  client_secret   = "YOUR_CLIENT_SECRET"
  tenant_id       = "YOUR_TENANT_ID"
  subscription_id  = "YOUR_SUBSCRIPTION_ID"
  resource_group_name = "YOUR_CUSTOM_RESOURCE_GROUP"
  image_name      = "YOUR_IMAGE_NAME"
  location        = "YOUR_LOCATION"
  }
